src/Views/HardwareView.vala

```diff
@@ -280,6 +282,60 @@ public class About.HardwareView : Gtk.Grid {
 
         return 0;
     }
+
+    private string get_storage_type () {
+        string partition_name = get_partition_name ();
+        string disk_name = get_disk_name (partition_name);
+        string path = "/sys/block/%s/queue/rotational".printf (disk_name);
+        string storage = "";
+        try {
+            string contents;
+            FileUtils.get_contents (path, out contents);
+            if (int.parse (contents) == 0) {
+                storage = _("(SSD)");
+            } else {
+                storage = _("(HDD)");
+            }
+        } catch (FileError e) {
+            warning (e.message);
+        }
+        return storage;
+    }
+
+    private string get_partition_name () {
+        string df_stdout;
+        string partition = "";
+        try {
+            Process.spawn_command_line_sync ("df /",
+                out df_stdout);
+            string[] output = df_stdout.split ("\n");
+            foreach (string line in output) {
+                if (line.has_prefix ("/dev/")) {
+                    int idx = line.index_of (" ");
+                    if (idx != -1) {
+                        partition = line.substring (0, idx);
```

Comment:
If you determined the partition, you can break out of the loop here, or even faster, just return it.

Reply:
Done :)